content/en/synthetics/guide/browser-tests-using-shadow-dom.md

```diff
@@ -0,0 +1,118 @@
+
+{{< img src="synthetics/guide/browser-tests-using-shadow-dom/open-shadow-dom.png" alt="Open Shadow DOM" style="width:50%;" >}}
+
+In `open` mode, you can use JavaScript assertions to interact with and validate elements rendered in a shadow DOM with the `Element.shadowRoot` property. 
+
+### Assert text presence
+
+{{< img src="synthetics/guide/browser-tests-using-shadow-dom/validate-text-in-shadow-dom.png" alt="Validate text rendered in a shadow DOM" style="width:90%;" >}}
+
+To validate that the text "TODO" appears on a page, query the `innerHTML` property directly from the `<body>` element of the main document.
+
+```HTML
+return document.querySelector("body").innerHTML.includes("TODO")
+```
+
+### Validate rendered text
+
+To validate that text is rendered within a given element rendered in a shadow DOM, use the `shadowRoot` property to access the respective element and `innerHTML` or `textContent` properties to validate the text is rendered.
+
+For example, the following code snippet validates the text "TODO" rendered in an `<h3>` HTML tag:
+
+```
+// find element to which the Shadow DOM is attached:
+let element = document.querySelector("body > editable-list")
+ 
+// use shadowRoot property to locate the <h3> element in the Shadow DOM:
+let shadowDomElement = element.shadowRoot.querySelector("div > h3")
+ 
+// check textContent of the Shadow DOM element:
+return shadowDomElement.textContent.includes("TODO")
+```
+
+### Confirm entered text
+
+To enter text into a text `<input>` field rendered in a shadow DOM, locate the `<input>` element and set the `value` field.
+
+{{< img src="synthetics/guide/browser-tests-using-shadow-dom/validate-text-type.png" alt="Validate entered text rendered in a shadow DOM" style="width:90%;" >}}
+
+For example, the following code snippet validates that the text "Item added with JS assertion" is added in the input field:
+
+```
+// find element to which the Shadow DOM is attached:
+let element = document.querySelector("body > editable-list")
+ 
+// use shadowRoot property to locate the <input> element in the Shadow DOM:
+let shadowDomInput = element.shadowRoot.querySelector("input")
+ 
+// set the value property of the <input> element:
+shadowDomInput.value = "Item added with JS assertion"
+ 
+return true
+```
+
+### Click on an element
+
+To trigger a click on an element rendered in a shadow DOM, locate the respective element and run `.click()` on it.
+
+{{< img src="synthetics/guide/browser-tests-using-shadow-dom/validate-trigger-click.png" alt="Validate triggered click on an element rendered in a shadow DOM" style="width:80%;" >}}
+
+For example, the following code snippet triggers a click on a button element.
+
+```
+// find element to which the Shadow DOM is attached:
+let element = document.querySelector("body > editable-list")
+ 
+// use shadowRoot property to locate the <button> element in the Shadow DOM:
+let shadowDomButton = element.shadowRoot.querySelector("button.editable-list-add-item")
```

Comment:
So the element here `<button class=editable-list-add-item icon>` has multiple classnames (separated by the space) 

In this case, querying by the `editable-list-add-item` class is enough to locate the element, but if you wanted to use both in your query, it would look something like `element.shadowRoot.querySelector("button.editable-list-add-item.icon")`.

For the purpose of this guide, I think the example query is fine as is (users will need to write their own queries specific to the site they're testing)

Reply:
I learned something 
thanks :D